### PR TITLE
Make `Checksum` task compatible with Gradle 5.0

### PR DESCRIPTION
**Pull request by bamboo**
By ignoring directories found in `IncrementalTaskInputs`.

This is necessary because Gradle 5.0 includes directories in `IncrementalTaskInputs` when the input is represented as a file tree.

### CODE REVIEW

File: src/main/java/org/gradle/crypto/checksum/Checksum.java [hunk 1]
```diff
@@ -116,7 +119,11 @@ public void execute(InputFileDetails inputFileDetails) {
         inputs.removed(new Action<InputFileDetails>() {
             @Override
             public void execute(InputFileDetails inputFileDetails) {
-                getProject().delete(outputFileFor(inputFileDetails.getFile()));
+                File input = inputFileDetails.getFile();
+                if (input.isDirectory()) {
```
Comment (wolfs) [inline src/main/java/org/gradle/crypto/checksum/Checksum.java:123]: We could delete removed directories, right? So we don't have empty directories lying around in the output directory.

I am not feeling strongly about this and I guess removing empty directories would also require extra coverage.

File: src/functionalTest/groovy/org/gradle/crypto/checksum/IncrementalChecksumSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            
+            task checksums(type: Checksum) {
+                files = fileTree("input") {
+                    include "**/*.txt"
+                }
+                outputDir = file("checksums")
+            }
+        """
+        newFolder('input')
+        newFile('input/foo.txt') << 'foo'
```
Comment (wolfs) [inline src/functionalTest/groovy/org/gradle/crypto/checksum/IncrementalChecksumSpec.groovy:36]: Should we add some files deeper in the directory structure? Note that even with pre-5.0 Gradle versions, the build probably would have failed, given that non-root directories have been reported as changes before as well.

**Review by wolfs — APPROVED** [review]
Looks good. I added two comments for further fixes/features. Feel free to address them or open issues for them.

File: src/functionalTest/groovy/org/gradle/crypto/checksum/IncrementalChecksumSpec.groovy (hunk 2 above)
Comment (bamboo) [reply src/functionalTest/groovy/org/gradle/crypto/checksum/IncrementalChecksumSpec.groovy:36]: Makes sense. Thanks.

File: src/main/java/org/gradle/crypto/checksum/Checksum.java (hunk 1 above)
Comment (bamboo) [reply src/main/java/org/gradle/crypto/checksum/Checksum.java:123]: The plot thickens. The plugin doesn't actually replicate the input directory tree in the output directory so there are no directories to remove.

**Pull request merged**